tests/integration/mixer/outboundtrafficpolicy/helper.go

```diff
@@ -309,32 +278,103 @@ func RunExternalRequestTest(expected map[string][]string, t *testing.T) {
 							},
 						})
 
-						key := tc.portName
 						if tc.gateway {
 							key += "_egress"
 						}
-						if err != nil && len(expected[key]) != 0 {
+
+						if err != nil && len(expected[key].Code) != 0 {
```

Comment:
Add a comment here to say that this is only useful for "BlackHole TCP case" where you don't get any response codes.

Otherwise, it's confusing for maintainers as to why we are ignoring errors.

Reply:
@howardjohn 

```go
if tc.gateway {
  key += "_egress"
}
```

is super hacky and makes it impossible to maintain.